configuration/pih/scripts/global/zl.js

```diff
@@ -88,3 +88,69 @@ function setUpExpandableContacts(badPhoneNumberMsg) {
 
 }
 
+function setUpDatepickerStartAndEndDateValidation(badDateInTheFutureMsg,badStartDateGreaterThanEndDateMsg){
+
+  jq(".startDateEndDate").each(function(j, domEl){
+    for(let i=0; i<jq(domEl).find('input[type=text]').length; i++){
+      if(i % 2 === 0){
+         jq('#startDate').prop("id", `startDate-${j}`);
+        jq('#endDate').prop("id", `endDate-${j}`);
+        setStartDateIdAndEndDateId(`startDate-${j}`,`endDate-${j}`)
+      }
+    }
+  })
+
+function setStartDateIdAndEndDateId(startDateId,endDateId) {
+
+    jq(`#${startDateId}`).change(function (e) {
```

Comment:
This doesn't make sense for a few reasons (though it may work). Let's instead try

```suggestion
  jq(".startDateEndDate").each(function(j, domEl){
    setUpDatepicker(this);
  });

  function setUpDatepickers(containerNode) {
    const startDatepicker = jq(jq(containerNode).find('.startDate'));
    const endDatepicker = jq(jq(containerNode).find('.endDate'));
    
    startDatepicker.change(function (e) {
```

etc?

Reply:
Thanks so much @brandones